app/src/screens/Splash.tsx

```diff
@@ -185,179 +109,43 @@ const Splash = () => {
-        if (!store.loginAttempt.lockoutDate) {
-          navigation.dispatch(
-            CommonActions.reset({
-              index: 0,
-              routes: [{ name: Screens.EnterPIN }],
-            })
-          )
-        } else {
-          // return to lockout screen if lockout date is set
-          navigation.dispatch(
-            CommonActions.reset({
-              index: 0,
-              routes: [{ name: Screens.AttemptLockout }],
-            })
-          )
-        }
-
-        return
-      }
-
-      // If onboarding was interrupted we need to pickup from where we left off.
-      navigation.dispatch(
-        CommonActions.reset({
-          index: 0,
-          routes: [
-            {
-              name: resumeOnboardingAt(
-                store.onboarding.didSeePreface,
-                store.onboarding.didCompleteTutorial,
-                store.onboarding.didAgreeToTerms,
-                store.onboarding.didCreatePIN,
-                store.onboarding.didNameWallet,
-                store.onboarding.didConsiderBiometry,
-                store.onboarding.didConsiderPushNotifications,
-                TermsVersion,
-                store.preferences.enableWalletNaming,
-                showPreface,
-                !!enablePushNotifications
-              ),
-            },
-          ],
-        })
-      )
-    } catch (e: unknown) {
-      setInitErrorType(InitErrorTypes.Onboarding)
-      setInitError(new BifoldError(t('Error.Title2030'), t('Error.Message2030'), (e as Error)?.message, 2030))
+    if (initializing.current) {
+      return
     }
-  }, [
-    mounted,
-    setStep,
-    store.authentication.didAuthenticate,
-    store.stateLoaded,
-    store.onboarding.onboardingVersion,
-    store.onboarding.didCompleteOnboarding,
-    store.onboarding.didSeePreface,
-    store.onboarding.didCompleteTutorial,
-    store.onboarding.didAgreeToTerms,
-    store.onboarding.didCreatePIN,
-    store.onboarding.didConsiderPushNotifications,
-    store.onboarding.didNameWallet,
-    store.onboarding.didConsiderBiometry,
-    store.preferences.enableWalletNaming,
-    enablePushNotifications,
-    showPreface,
-    dispatch,
-    store.loginAttempt.lockoutDate,
-    navigation,
-    initOnboardingCount,
-    t,
-  ])
 
-  useEffect(() => {
+    setStep(1)
+    setStep(2)
+
     const initAgentAsyncEffect = async (): Promise<void> => {
       try {
-        if (
-          !mounted ||
-          initializing.current ||
-          !store.authentication.didAuthenticate ||
-          !store.onboarding.didConsiderBiometry ||
-          store.onboarding.postAuthScreens.length > 0
-        ) {
-          return
-        }
-
-        initializing.current = true
-
         setStep(3)
         await (ocaBundleResolver as RemoteOCABundleResolver).checkForUpdates?.()
 
         setStep(4)
         const agent = await initializeAgent()
+        initializing.current = true
+
         if (!agent) {
-          initializing.current = false
```

Comment:
I think we still need these two, no? Or else if it fails initialization the first, the retry button won't work

Reply:
Fixed